File: AndroidProject/CollectorAgentMind/src/main/java/com/example/collectoragentmind/codelets/behaviors/Forage.java
package com.example.collectoragentmind.codelets.behaviors;

import com.example.collectoragentmind.util.Apple;

import java.util.ArrayList;

import br.unicamp.cst.core.entities.Codelet;
import br.unicamp.cst.core.entities.MemoryObject;

public class Forage extends Codelet {

    /*
        If there are no known apples, makes the legsMO
        null
    */

    private MemoryObject knownApplesMO;
    private MemoryObject legsMO;

    @Override
    public void accessMemoryObjects() {
        knownApplesMO = (MemoryObject) this.getInput("KNOWN_APPLES");
        legsMO = (MemoryObject) this.getOutput("LEGS");
    }

    @Override
    public void calculateActivation() {

    }

    @Override
    public void proc() {
        ArrayList<Apple> knownApples = (ArrayList<Apple>) knownApplesMO.getI();

        if (knownApples.size() == 0) {
            legsMO.setI(null);
        }
    }
}

File: GodotKotlin/CoreModelTrail/src/main/kotlin/com/example/game/cst/codelets/behaviors/Forage.kt
package com.example.game.cst.codelets.behaviors

import br.unicamp.cst.core.entities.Codelet
import br.unicamp.cst.core.entities.MemoryObject
import com.example.game.environment.Apple

class Forage : Codelet() {
    /*
        If there are no known apples, makes the legsMO
        null
    */

    private var knownApplesMO: MemoryObject? = null
    private var legsMO: MemoryObject? = null

    override fun accessMemoryObjects() {
        knownApplesMO = getInput("KNOWN_APPLES") as MemoryObject
        legsMO = getOutput("LEGS") as MemoryObject
    }

    override fun calculateActivation() {}

    override fun proc() {
        val knownApples: ArrayList<Apple> = knownApplesMO!!.i as ArrayList<Apple>
        if (knownApples.size == 0) {
            legsMO?.i = null
        }
    }
}